### PR TITLE
Check formatting and Clippy lints in CI

### PR DESCRIPTION
**Pull request by djc**
Stacked on the branch for #519 in order to clean up clippy lints in the benchmarks.

### CODE REVIEW

File: html5ever/src/tokenizer/interface.rs [hunk 1]
```diff
@@ -27,6 +27,12 @@ pub struct Doctype {
...
 }
 
+impl Default for Doctype {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl Doctype {
     pub fn new() -> Doctype {
```
Comment (mrobinson) [inline html5ever/src/tokenizer/interface.rs:37]: We should likely remove `Doctype::new()` now and change all references to use the new `Default` implementation.

File: xml5ever/src/tokenizer/mod.rs [hunk 2]
```diff
@@ -1070,9 +1068,8 @@ impl<Sink: TokenSink> XmlTokenizer<Sink> {
             },
             //§ bogus_doctype_state
             XmlState::BogusDoctype => loop {
-                match get_char!(self, input) {
-                    '>' => go!(self: emit_doctype; to Data),
-                    _ => (),
+                if let '>' = get_char!(self, input) {
```
Comment (mrobinson) [inline xml5ever/src/tokenizer/mod.rs:1071]:
```suggestion
                if '>' == get_char!(self, input) {
```

File: markup5ever/util/buffer_queue.rs [hunk 3]
```diff
@@ -49,6 +49,12 @@ pub struct BufferQueue {
     buffers: VecDeque<StrTendril>,
 }
 
+impl Default for BufferQueue {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
```
Comment (mrobinson) [inline markup5ever/util/buffer_queue.rs:57]: Same comment here about removing the `new()` function.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Just a couple small comments:

File: html5ever/src/tokenizer/interface.rs (hunk 1 above)
Comment (djc) [reply html5ever/src/tokenizer/interface.rs]: Good call, done.

File: xml5ever/src/tokenizer/mod.rs (hunk 2 above)
Comment (djc) [reply xml5ever/src/tokenizer/mod.rs]: Heh, yup.

File: markup5ever/util/buffer_queue.rs (hunk 3 above)
Comment (djc) [reply markup5ever/util/buffer_queue.rs]: Done.

**Comment by djc** [comment]
(Cleaned up clippy suggestions in rcdom tests/examples after your review.)

**Review by mrobinson — APPROVED** [review]

**Pull request merged**